src/main/java/org/oldbox/clippy/ClippyRepository.java

```diff
@@ -67,6 +68,34 @@ public void addEntryToCategory(String categoryName, NoteEntry entry) throws File
         return this.getDatabase().keySet();
     }
 
+    public List<Category> getCategories(List<String> categoryNames) throws FileSystemException {
+        List<Category> categories = new ArrayList<>();
+        List<String> failedNames = new ArrayList<>();
+        for (String name : categoryNames) {
+            try {
+                categories.add(getCategory(name));
+            } catch (FileSystemException e) {
+                failedNames.add(name);
+            }
+        }
+
+        if(failedNames.size() > 0) {
+            StringBuilder builder = getFailedCategoryNamesMessage(failedNames);
+            throw new FileSystemException(builder.toString());
+        }
+
+        return categories;
+    }
+
+    private StringBuilder getFailedCategoryNamesMessage(List<String> failedNames) {
```

Comment:
Some points about this:

 - I like that you moved the whole thing to a method
 - The method could be static
 - Should return the string and not the builder.

Reply:
Very good point! Feeling silly that I didn't realise that before...